### PR TITLE
Refactor commonJS to ES6

### PR DESCRIPTION
**Pull request by jsvoo**
Converted commonJS files to ES6 modules, changed require("") statements to import statements, modified commonJS-specific functions to work with the new ES6 modules, got rid of got.

### CODE REVIEW

File: app.mjs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    maxAge: "600"
+  })
+)
+app.use(logger("dev"))
+app.use(express.json())
+app.use(express.urlencoded({extended: true}))
+app.use(cookieParser())
+
+//Publicly available scripts, CSS, and HTML pages.
+app.use(express.static(path.join(__dirname, "public")))
```
Comment (cubap) [inline app.mjs:68]: There isn't a public directory, so this isn't strictly useful, but maybe it'll make certs easier.

File: app.mjs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+app.use("/", indexRouter)
+
+//catch 404 because of an invalid site path
+app.use(function (req, res, next) {
+  let msg = res.statusMessage ?? "This page does not exist"
+  res.status(404).send(msg)
+  res.end()
```
Comment (cubap) [inline app.mjs:76]: I'm pretty sure `send()` closes the response and `.end()` is not required.

Comment (cubap) [reply app.mjs:76]: (acknowledging that this is the old file and not your handiwork)

File: routes/tpen-users.mjs [hunk 3]
```diff
@@ -1,9 +1,17 @@
...
+import express from "express"
+import path from "path"
 const router = express.Router()
-const managementRouter = require("./manage-api.js")
+import managementRouter from "./manage-api.mjs"
+
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
```
Comment (cubap) [inline routes/tpen-users.mjs:11]: This part look like it was frustrating to figure out.

File: app.mjs (hunk 1 above)
Comment (jsvoo) [reply app.mjs:68]: image


there is this public directory that holds the script and other stuff in it. Are we taking those out?

File: app.mjs (hunk 2 above)
Comment (jsvoo) [reply app.mjs:76]: Yea, I missed that. I'd go ahead to change those to json instead. and yes, end() is not necessary

File: routes/tpen-users.mjs (hunk 3 above)
Comment (jsvoo) [reply routes/tpen-users.mjs:11]: it was. It's even worse that this had to be repeated across all files that needed to reference public

File: public/script/tpen_public_auth.js [hunk 4]
```diff
@@ -17,13 +17,15 @@ const ISSUER_BASE_URL = "cubap.auth0.com"
 const CLIENT_ID = "bBugFMWHUo1OhnSZMpYUXxi3Y1UJI7Kl"
 const DOMAIN = "cubap.auth0.com"
 
+const returnTo = "https://three.t-pen.org/callback"
```
Comment (cubap) [inline public/script/tpen_public_auth.js:20]: This is a good way to do it. We can look at a separate page for logout later, if that's useful.

**Review by cubap — APPROVED** [review]
Well done. I'll push it so you have it and if @thehabes finds something in testing, we'll just hotfix it.

**Comment by thehabes** [comment]
Yeaaaaaaaaah I'll let you know.  We need to get those tests to pass.

**Pull request merged**